### ISSUE TITLE
Installation in $HOME broke KDE Plasma's icons symlink

### ISSUE THREAD

**Issue by JorgeCepeda** [noisetorch/NoiseTorch#335]
**Describe the bug**
Installing NoiseTorch in $HOME made its icons folder replace the icons folder from KDE Plasma in ~/.local/share/icons

**To Reproduce**
1. Have KDE Plasma installed
2. Install NoiseTorch on $HOME

**Expected behavior**
NoiseTorch's installation shouldn't break existing files/directories

**Screenshots**
![image](https://user-images.githubusercontent.com/69733604/174190357-8ed33b37-6a41-4d1d-b316-3778aaf883d5.png)
Left picture is the current one, without symlink, right is a backup with what was there, containing KDE themes

**Logs**

**Desktop (please complete the following information):**
 - Distribution [e.g. Ubuntu 22.04, Arch]: Feren OS
 - DE [e.g. Plasma 5.24.5, GNOME 42.0]: 5.24.5
 - Pulseaudio/Pipewire Version: 13.99.1
 - NoiseTorch Version [e.g. v0.12.0]: v0.12.1

**Additional context**

**Comment by RiemaruKarurosu** [comment]
I can't reproduce it in Manjaro Kde




![Screenshot_20220616_204636](https://user-images.githubusercontent.com/45057400/174206060-98f7c5dc-2ddf-4602-8d20-55ed675d9cc4.png)
![Screenshot_20220616_204725](https://user-images.githubusercontent.com/45057400/174206166-aa6ccd89-33c5-4538-8e7a-03e7e9f5d09e.png)

**Comment by TheDukeofErl** [comment]
Hm, on my system (plain old Arch) icons is a directory, not a symlink, and the only folder I have in there is the hicolor folder. 

That said, I don't use any custom themes, just plain old Breeze, so that might have something to do with it. Have both of you installed local themes as user then, I'm guessing?

**Comment by JorgeCepeda** [comment]
Themes were added through the Themes menu, as user probably

**Comment by ZyanKLee** [comment]
Thanks for reporting this issue.

We could think about installing to `~/.var/app/com.github.noisetorch.NoiseTorch` (similar to how EasyEffects does) and then symlink files into the corresponding folders in `~/.local/`. WDYT?

**Comment by JorgeCepeda** [comment]
I like the symlink idea better, at least in my case that symlink points to ~/.icons and there are apps that put theirs there in hicolor

**Comment by TheDukeofErl** [comment]
> We could think about installing to another directory structure and then symlink files into the corresponding folders in ~/.local/.

I don't believe this would actually fix the problem in this case: even if ~/.local/share/icons is a symlink I believe the unpack of the tar file should be fine and put the files in the right place.

@JorgeCepeda can you please read the link that you have at ~/.local/share/icons? I have a hunch that it's going to go to a location owned by root.

**Comment by JorgeCepeda** [comment]
> > We could think about installing to another directory structure and then symlink files into the corresponding folders in ~/.local/.
> 
> I don't believe this would actually fix the problem in this case: even if ~/.local/share/icons is a symlink I believe the unpack of the tar file should be fine and put the files in the right place.
> 
> @JorgeCepeda can you please read the link that you have at ~/.local/share/icons? I have a hunch that it's going to go to a location owned by root.

It seems we commented at the same time, answered it

**Comment by JorgeCepeda** [comment]
A search on tar unpacking replacing symlinks says the -h flag would preserve them

**Comment by TheDukeofErl** [comment]
> symlink points to ~/.icons

Interesting, that's the legacy location per [the XDG spec](https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html#directory_layout). I'm kind of surprised that FerenOS is set up that way, but that probably explains why we haven't seen this issue til now.

> replacing symlinks says the -h flag would preserve them

Yup, `tar -C $HOME -h -xzf Noisetorch_x64_v0.12.2.tgz` does the trick and shouldn't cause any issues on systems not using symlinks. I'll update the readme.

**Comment by TheDukeofErl** [comment]
https://github.com/noisetorch/NoiseTorch/pull/336 should resolve this. Looking at https://github.com/noisetorch/NoiseTorch/blob/master/untar.go, it doesn't look like the updater will have this problem, as it appears to be building the paths and copying the files out of the tarball.

**Comment by JorgeCepeda** [comment]
I just booted up, yesterday I unpacked the new version with -h, the icons folder symlink now became a normal folder with hicolor and NoiseTorch's icon

**Comment by JorgeCepeda** [comment]
I tried repeating some of the steps that lead to this but it hasn't happened again
1. Install v.12.2 over v.12.1 (this time I reinstalled completely)
2. Reboot and try it
3. Set discord to start automatically
4. Shutdown, boot Windows 11
5. 2 hours of dealing with Windows Update (not done again)
6. Boot Feren OS, symlink gone

**Comment by TheDukeofErl** [comment]
I'm not sure why you would be seeing any issue with the updated install instructions, I tested it with FerenOS 2022.04 in a VM and didn't have any problems.

**Comment by Technetium1** [comment]
If this can't be reproduced on Feren OS I think it should probably be closed then?

**Comment by JorgeCepeda** [comment]
I guess so, the only thing that comes to mind is some race condition where noisetorch tries to verify its data and fails but it doesn't make much sense since all data is there

**Comment by TheDukeofErl** [comment]
Going to go ahead and close this.
